crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/stateless/StatelessAssessmentService.java

```diff
@@ -34,97 +31,92 @@ public StatelessAssessmentService(AssessmentCriteriaService assessmentCriteriaSe
                                       FullAssessmentAvailabilityService fullAssessmentAvailabilityService) {
 
         super(assessmentCriteriaService);
-        this.assessmentCriteriaService = assessmentCriteriaService;
         this.meansAssessmentServiceFactory = meansAssessmentServiceFactory;
         this.fullAssessmentAvailabilityService = fullAssessmentAvailabilityService;
     }
 
     public StatelessResult execute(Assessment assessment,
                                    Map<AgeRange, Integer> childGroupings,
                                    List<Income> income,
-                                   List<Outgoing> outgoings) {
+                                   List<Outgoing> outgoings,
+                                   AssessmentCriteriaEntity assessmentCriteriaEntity) {
+        var initialAnswer = initialResult(
+                childGroupings,
+                assessmentCriteriaEntity,
+                assessment.getCaseType(),
+                assessment.getMagistrateCourtOutcome(),
+                income,
+                null);
 
         if (assessment.getAssessmentType() == StatelessRequestType.INITIAL) {
-            return initialOnly(assessment, childGroupings, income);
+            return new StatelessResult(null, initialAnswer);
         } else {
-            return initialAndFull(assessment, childGroupings, income, outgoings);
+            if (initialAnswer.isFullAssessmentPossible()) {
+                final var statelessFullResult =
+                        fullResult(childGroupings, assessmentCriteriaEntity, assessment.getEligibilityCheckRequired(), assessment.getCaseType(), outgoings,
+                                initialAnswer.getTotalAggregatedIncome());
+                return new StatelessResult(statelessFullResult, initialAnswer);
+            } else {
+                return new StatelessResult(null, initialAnswer);
+            }
         }
     }
 
-    private StatelessResult initialAndFull(@NotNull Assessment assessment,
-                                           @NotNull Map<AgeRange, Integer> childGroupings,
-                                           @NotNull List<Income> incomes,
-                                           @NotNull List<Outgoing> outgoings) {
+    public StatelessFullResult fullResult(Map<AgeRange, Integer> childGroupings,
+                                           AssessmentCriteriaEntity criteriaEntry,
+                                           boolean eligibilityCheckRequired,
+                                           CaseType caseType,
+                                           List<Outgoing> outgoings,
+                                           BigDecimal totalIncome) {
```

Comment:
Original comment from @Mhowell494: 
> For me, this is too many parameters, however, there is a degree of preference involved. That said, is there a way we could reduce this?For example, moving the mapChildGroupings call? A lot are being passed in just to build the MeansAssessmentRequestDTO, can't we just build this in the execute method instead?

Reply:
With this proposal I see us replacing 4 parameters (childGrouping, criteriaEntry, totalIncome, eligibilityCheckRequired) with a DTO containing 29 fields. I think 25 unused fields just adds confusion, and you lose a key benefit of abstracting this functionality into a separate method.

I can see DTOs are helpful when you want to batch up a lot of fields, but I read that it is a known gotcha to use them in sitatuations when most of the fields are not in use - see https://www.baeldung.com/java-dto-pattern#common-mistakes

Reply:
I'm not a fan of lots of input parameters however I agree that using a DTO with 25 unused fields is not ideal, also creating a new DTO specifically for this might also be overkill. I'm happy to resolve this comment for now.